src/components/Alert/AlertContainer.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Alert from 'react-alert';
+
+import { updateThershold, updateInterval } from './../../store/actions';
+import {
+    selectCurrentInterval,
+    selectUptimes,
+    selectCurrentTimerCount
+} from './../../store/selectors';
+
+import { currentAverage } from '../../utils/intervals';
+import {
+    TIME_LIMIT,
+    UPTIME_FECTH_INTERVAL,
+    AVERAGE_UPTIME_INTERVAL
+} from '../../utils/constants';
+import { ALERT_OPTIONS } from './alertOptions';
+import { errorMessage, sucessMessage } from './alertMessages';
+
+class AlertContainer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            errorHasOccured: false
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const { currentTimerCount } = this.props;
+
+        if (
+            nextProps.currentTimerCount !== currentTimerCount &&
+            currentTimerCount % UPTIME_FECTH_INTERVAL === 0
+        ) {
+            this.fireAlert(currentTimerCount);
+        }
+
+        if (
+            nextProps.currentTimerCount !== currentTimerCount &&
+            currentTimerCount % AVERAGE_UPTIME_INTERVAL === 0
+        ) {
+            //every two minutes change the interval
+            debugger
+
+            this.props.updateInterval();
+        }
+    }
+
+    fireAlert(time) {
+        const { average } = this.props;
+        this.props.updateThershold(average, time);
+
+        if (average > 1) {
+            this.fireHighLoadAlert();
+        }
+
+        if (this.state.errorHasOccured && average < 1) {
+            this.fireRecoveryAlert();
+        }
+    }
+
+    fireHighLoadAlert() {
+        this.showAlert('error', this.props.average);
+        this.setState({ errorHasOccured: true });
+    }
+
+    fireRecoveryAlert() {
+        this.showAlert('success', this.props.average);
+        this.setState({ errorHasOccured: false });
+    }
+
+    message(type) {
```

Comment:
On line 82 you call this message function with two parameters but it is only defined to accept one.

Also within the message function you are deconstructing `average` from the props, but calling the message function on line 82 with `props.average`.

This is just my personal preference but I try to avoid adding methods to containers if I can and it looks like you can here with the `message` function

e.g

````javascript
const message = (type, currentTimerCount, average) => type === 'success' ? sucessMessage(average, currentTimerCount) : errorMessage(average, currentTimerCount);
````